### DISCUSSION TITLE
Absolute beginner to programming, how do I use this

### DISCUSSION THREAD

**Discussion by donel12** [BepInEx/BepInEx#791] [General]
I'm in year 12 studying computer science as one of my a levels, and want to begin doing projects as soon as possible. I've installed PyCharm for my choice of an IDE, and will be coding in python.

The first thing I noticed about this is that it creates multiple folders and files for itself when all I'm trying to do is create a single program for testing out some very basic functions like sorting and searching algorithms (yes, I'm a newbie).

I've heard uploading all your projects to GitHub is recommended as it allows you to track your progress, backup work and start building a portfolio all at the same time but I'm struggling to understand which of these files I need to upload and how to go about doing so.

Previously, I was accustomed to seeing .py standalone executables that I can edit with something simple like IDLE or Notepad++, not any of this multi-file/folder stuff.

Any advice would be greatly appreciated.

**Comment by ManlyMarco** [comment]
Read https://www.freecodecamp.org/news/introduction-to-git-and-github/ and https://docs.github.com/en/get-started/using-git for using git and GitHub.
